src/control/cmd/dmg/firmware.go

```diff
@@ -138,16 +158,50 @@ func (cmd *firmwareUpdateCmd) Execute(args []string) error {
 	if err := control.PrintResponseErrors(resp, &bld); err != nil {
 		return err
 	}
-	if cmd.Verbose {
-		if err := pretty.PrintSCMFirmwareUpdateMapVerbose(resp.HostSCMResult, &bld); err != nil {
+
+	if cmd.isSCMUpdate() {
+		if err := cmd.printSCMUpdateResult(resp, &bld); err != nil {
 			return err
 		}
 	} else {
-		if err := pretty.PrintSCMFirmwareUpdateMap(resp.HostSCMResult, &bld); err != nil {
+		if err := cmd.printNVMeUpdateResult(resp, &bld); err != nil {
 			return err
 		}
 	}
+
 	cmd.log.Info(bld.String())
 
 	return resp.Errors()
 }
+
+func (cmd *firmwareUpdateCmd) isSCMUpdate() bool {
+	return cmd.DeviceType == "scm"
+}
+
+func (cmd *firmwareUpdateCmd) printSCMUpdateResult(resp *control.FirmwareUpdateResp, out io.Writer) error {
+	if cmd.Verbose {
+		if err := pretty.PrintSCMFirmwareUpdateMapVerbose(resp.HostSCMResult, out); err != nil {
+			return err
+		}
+	} else {
```

Comment:
return early to avoid the else?

Reply:
Good idea, I can clean this up a bit. I'll add the change once your PR lands, since I'm sure I'll need to resolve some things in the merge.